src/styles/marg.js

```diff
@@ -10,47 +10,37 @@ const marg = ( props ) => {
     // see cyverse-ui/src/style/README.md
     //
 
+    const returnValue = (value, sizes) => {
+        if (typeof value == "number") {
+            return value === 0 ? 0 : sizes[value -1]
+        } else { 
+            return value
+        }
+    };
+
     return Object.keys(props).reduce((style, prop) => {
         let value = props[prop];
 
         let cssField;
         switch (prop) {
             case "mr":
-                style.marginRight = typeof value == "number"
-                    ? sizes[value - 1]
-                    : value
+                style.marginRight = returnValue(value, sizes);
                 break;
             case "mb":
-                style.marginBottom = typeof value == "number"
-                    ? sizes[value - 1]
-                    : value
+                style.marginBottom = returnValue(value, sizes);
                 break;
             case "ml":
-                style.marginLeft = typeof value == "number"
-                    ? sizes[value - 1]
-                    : value
+                style.marginLeft = returnValue(value, sizes);
                 break;
             case "ms":
-                style.marginRight = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
-                style.marginLeft = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
+                style.marginRight = returnValue(value, sizes);
+                style.marginLeft = returnValue(value, sizes);
                 break;
             case "m":
-                style.marginTop = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
-                style.marginRight = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
-                style.marginBottom = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
-                style.marginLeft = typeof value == "number"
-                    ? sizes[value -1]
-                    : value
+                style.marginTop = returnValue(value, sizes);
+                style.marginRight = returnValue(value, sizes);
+                style.marginBottom = returnValue(value, sizes);
+                style.marginLeft = returnValue(value, sizes);
```

Comment:
Unfortunately, the terse example here doesn't evaluate the way it is intended because behaviour of switch is that not supplying `break` causes the next statement to execute regardless of match. Which in retrospect is why you can do
```
case "ms":
case "mr:
style.marginRight = ...
```
But then without `break` following it continues to execute the rest of the assignments. We can't however  place a break after the assignments because it would then jump out and miss the other parts to the object we are building.